apis/python/src/tiledbsoma/_collection.py

```diff
@@ -509,9 +509,48 @@ class Collection(
     CollectionBase[CollectionElementType], somacore.Collection[CollectionElementType]
 ):
     """
-    A persistent collection of SOMA objects, mapping string keys to any SOMA object.
+    ``Collection`` is a persistent collection of named SOMA objects, stored as
+    a mapping of string keys and SOMA object values. Values may be any persistent
+    ``tiledbsoma`` object, including DataFrame, SparseNDArray, DenseNDArray,
```

Comment:
```
``DataFrame``
```
etc -- with backticking since these are code names

you do this some places below -- let's simply be consistent

(alternative: `<code>DataFrame</code>` etc.)